src/RageSoundMixBuffer.cpp

```diff
@@ -4,17 +4,30 @@
 
 #include <cmath>
 #include <cstdint>
+#include <cstdlib>
 
 RageSoundMixBuffer::RageSoundMixBuffer()
 {
-	m_iBufSize = m_iBufUsed = 0;
-	m_pMixbuf = nullptr;
+	// Set m_iBufSize to 2MB.
+	m_iBufSize = 2 * 1024 * 1024 / sizeof(float);
+
+	// Allocate memory
+	m_pMixbuf = static_cast<float*>(std::malloc(m_iBufSize * sizeof(float)));
+
+	// Check if memory allocation was successful
+	if (m_pMixbuf == nullptr) {
+		ASSERT_M(false, "Sound mixing buffer being null should not be possible.");
```

Comment:
I'd change the message to "Failed to allocate memory for the sound mixing buffer"